### ISSUE TITLE
`PolySlab.intersections_plane()` returns wrong results if slab bounds are inf on one side

### ISSUE THREAD

**Issue by momchil-flex** [flexcompute/tidy3d#1539]
**Describe the bug**
`PolySlab.intersections_plane()` can return a None structure if slab bounds are inf on one side, which can cause an error in the solver. Similarly it can also wrongly return a shape with inf bounds on both sides. Something needs to be fixed [around here](https://github.com/flexcompute/tidy3d/blob/453055e89dcff6d619597120b47817e996f1c198/tidy3d/components/geometry/polyslab.py#L647-L664) but the method is not super clear so best whoever wrote it to have a look. Was it @weiliangjin2021 ?

**To Reproduce**

```
poly = td.PolySlab(
    vertices=[
       [2  , -0.15],
       [-2 , -0.15],
       [-2 ,  0.15],
       [2  ,  0.15]
    ],
    slab_bounds=[0, td.inf],
)
poly.intersections_plane(x=0)
```

Returns: [None]

```
poly = td.PolySlab(
    vertices=[
       [2  , -0.15],
       [-2 , -0.15],
       [-2 ,  0.15],
       [2  ,  0.15]
    ],
    slab_bounds=[-td.inf, 0],
)
poly.intersections_plane(x=0)
```

Returns: `[<POLYGON ((0.15 -Infinity, 0.15 Infinity, -0.15 Infinity, -0.15 -Infinity, 0...>]`

**Expected behavior**

In the first example, the return should be the correct shapely Polygon. In the second example, the Polygon should extend from -Infinity to 0, instead of from -Infinity to Infinity.

**Comment by tylerflex** [comment]
just a reminder to try to set a priority and size so we can track these more effectively. thanks

**Comment by momchil-flex** [comment]
Ah yeah I should have done that, but there's a PR fixing it already, I just need to review it quickly.

**Comment by momchil-flex** [comment]
Fixed in #1547
